### PR TITLE
Conver osc-cl1 secrets to use ESO.

### PR DESCRIPTION
**Pull request by HumairAK**
In this change apps superset and trino dev in osc-cl1 have their secrets
added to vault and removed from git, with external secrets taking their
place instead.

Furthers: https://github.com/operate-first/apps/issues/1996

### CODE REVIEW

File: kfdefs/overlays/osc/osc-cl1/trino-dev/kustomization.yaml [hunk 1]
```diff
@@ -3,10 +3,10 @@ apiVersion: kustomize.config.k8s.io/v1beta1
...
 resources:
+  - configs
+  - externalsecrets
+  - hive-metastores
   - kfdef.yaml
   - secure-route.yaml
-  - secrets
-  - hive-metastores
-  - configs
 generatorOptions:
```
Comment (Gregory-Pereira) [inline kfdefs/overlays/osc/osc-cl1/trino-dev/kustomization.yaml:11]: I thought generatorOptions pertained to secretGenerators. Does it also extend to configurations for configmapGenerators? If so it makes sense to keep this, but if not it should be deleted, no?

File: kfdefs/overlays/osc/osc-cl1/superset/externalsecrets/superset.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: external-secrets.io/v1beta1
```
Comment (Gregory-Pereira) [inline kfdefs/overlays/osc/osc-cl1/superset/externalsecrets/superset.yaml:1]: Should this fine not be named `superset-custom.yaml` to keep the filename and secret names the same?

**Review by Gregory-Pereira — COMMENTED** [review]
/lgtm, just two questions

File: kfdefs/overlays/osc/osc-cl1/trino-dev/kustomization.yaml (hunk 1 above)
Comment (HumairAK) [reply kfdefs/overlays/osc/osc-cl1/trino-dev/kustomization.yaml:11]: yes, you'll see configmapgenerator add a hash (config-maps are technically supposed to be immutable, as initially intended anyway), we use this option to disable it

**Review by erikerlandson — APPROVED** [review]

File: kfdefs/overlays/osc/osc-cl1/superset/externalsecrets/superset.yaml (hunk 2 above)
Comment (HumairAK) [reply kfdefs/overlays/osc/osc-cl1/superset/externalsecrets/superset.yaml:1]: yep yep tyty

File: kfdefs/overlays/osc/osc-cl1/trino-dev/kustomization.yaml (hunk 1 above)
Comment (Gregory-Pereira) [reply kfdefs/overlays/osc/osc-cl1/trino-dev/kustomization.yaml:11]: Thanks for the explanation! In that case ...
/lgtm
/approve

**Pull request merged**